src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -429,6 +441,9 @@ void reactOnTopologyChange() {
                 .toArray(Partition[]::new);
         addTask(() -> refreshTopologyUnlocked(assignedPartitions));
         trackIdleness(topology);
+        if (!streamInitialized) {
```

Comment:
could you elaborate on why we want to do it? I see on the previous lines 
442: `addTask(() -> refreshTopologyUnlocked(assignedPartitions));`
so we have already scheduled for execution

Reply:
Right, fixed this in `a95b29f`. Thanks for pointing out!

The idea was to force refreshing topology, once during the initialization (the partition assignment to this session is already scheduled by re-balance in the end of `StartingState` (as the session is registered)).